cocotb/binary.py

```diff
@@ -32,28 +32,61 @@
 
 import os
 import random
+import re
 import warnings
 
+
+_RESOLVE_TO_0      = "-lL"
+_RESOLVE_TO_1      = "hH"
+_RESOLVE_TO_CHOICE = "xXzZuUwW"
 resolve_x_to = os.getenv('COCOTB_RESOLVE_X', "VALUE_ERROR")
 
-def resolve(string):
-    for char in BinaryValue._resolve_to_0:
-        string = string.replace(char, "0")
-    for char in BinaryValue._resolve_to_1:
-        string = string.replace(char, "1")
-    for char in BinaryValue._resolve_to_error:
-        if resolve_x_to == "VALUE_ERROR" and char in string:
-            raise ValueError("Unable to resolve to binary >%s<" % string)
+
+class ResolveTable(dict):
+    """Translation table class for resolving binary strings.
+
+    For use with :func:`str.translate()`, which indexes into table with Unicode ordinals.
+    """
+    def __init__(self):
+        self.update({ord("0"): ord("0"), ord("1"): ord("1")})
+        self.update({ord(k): ord("0") for k in _RESOLVE_TO_0})
+        self.update({ord(k): ord("1") for k in _RESOLVE_TO_1})
```

Comment:
No need for `ord` I think

Reply:
`str.translate()` indexes into table with an ordinal, but you can return ordinals or strings.
This is faster than converting back to a string: 
```
return random.getrandbits(1) + 48
```
so I return ordinals for everything.
Also `str.maketrans()` produces a `dict` with ordinal->ordinal mapping.